src/utils/popoverMixins.js

```diff
@@ -100,34 +100,55 @@ export default {
     calculateOffset (trigger, popover) {
       switch (this.placement) {
         case 'top' :
-          this.position.left = trigger.offsetLeft - popover.offsetWidth / 2 + trigger.offsetWidth / 2
-          this.position.top = trigger.offsetTop - popover.offsetHeight
+          this.position.left = trigger.offsetLeft;
+          this.position.top = trigger.offsetTop - popover.offsetHeight;
           if (this.isPopover) {
             this.position.top -= this.$refs.arrow.offsetHeight;
           }
           break
         case 'left':
           this.position.left = trigger.offsetLeft - popover.offsetWidth
-          this.position.top = trigger.offsetTop + trigger.offsetHeight / 2 - popover.offsetHeight / 2
+          this.position.top = trigger.offsetTop
           if (this.isPopover) {
             this.position.left -= this.$refs.arrow.offsetWidth;
           }
           break
         case 'right':
-          this.position.left = trigger.offsetLeft + trigger.offsetWidth
-          this.position.top = trigger.offsetTop + trigger.offsetHeight / 2 - popover.offsetHeight / 2
+          const triggerClientRects = trigger.getClientRects();
+          if (triggerClientRects) {
+            this.position.left = trigger.offsetLeft + triggerClientRects[0].width
+          } else {
+            this.position.left = trigger.offsetLeft + trigger.offsetWidth
+          }
+          this.position.top = trigger.offsetTop
           break
         case 'bottom':
-          this.position.left = trigger.offsetLeft - popover.offsetWidth / 2 + trigger.offsetWidth / 2
+          this.position.left = trigger.offsetLeft
           this.position.top = trigger.offsetTop + trigger.offsetHeight
           break
         default:
           console.warn('Wrong placement prop')
       }
+
+      // don't let the popover go out of screen
+      const popoverViewportOffset = popover.getBoundingClientRect();
+      if (popoverViewportOffset.left < 0) {
+        this.position.left -= popoverViewportOffset.left;
+      }
+      if (popoverViewportOffset.top < 5) {
+        this.position.top -= popoverViewportOffset.top - 5;
+      }
+      if (popoverViewportOffset.left > window.innerWidth - popover.offsetWidth) {
+        this.position.left -= popoverViewportOffset.left - (window.innerWidth - popover.offsetWidth);
+      }
+      // TODO This offset does not work, it will just go off the screen
```

Comment:
@nusjzx As mentioned in person, this is the part of the code that does not work.

No rush into looking at this, you can focus on your other work first (this use case is not in `website-base`, so not a very big issue atm).

Thanks for taking a look at this!

Reply:
This part of code should be put after line 90, where all the adjustment are done. That will solve the problem.

Reply:
@nusjzx thanks for checking! The fix works now, will tidy up the PR.